Gems/Prefab/PrefabBuilder/PrefabGroup/DefaultProceduralPrefab.cpp

```diff
@@ -0,0 +1,553 @@
+#include <SceneAPI/SceneCore/DataTypes/GraphData/IMeshData.h>
+#include <SceneAPI/SceneCore/DataTypes/GraphData/ITransform.h>
+#include <SceneAPI/SceneData/Groups/MeshGroup.h>
+#include <SceneAPI/SceneData/Rules/CoordinateSystemRule.h>
+#include <SceneAPI/SceneData/Rules/LodRule.h>
+#include <AzCore/Component/EntityId.h>
+
+namespace AZ
+{
+    AZ_TYPE_INFO_SPECIALIZE(AZ::SceneAPI::PrefabGroupEvents::ManifestUpdates, "{B84CBFB5-4630-4484-AE69-A4155A8B0D9B}");
+}
+
+namespace AZ::SceneAPI
+{
+    struct PrefabGroupNotificationHandler final
+        : public AZ::SceneAPI::PrefabGroupNotificationBus::Handler
+        , public AZ::BehaviorEBusHandler
+    {
+        AZ_EBUS_BEHAVIOR_BINDER(
+            PrefabGroupNotificationHandler,
+            "{F1962BD1-D722-4C5F-A883-76F1004C3247}",
+            AZ::SystemAllocator,
+            OnUpdatePrefabEntity);
+
+        virtual ~PrefabGroupNotificationHandler() = default;
+
+        void OnUpdatePrefabEntity(const AZ::EntityId& prefabEntity) override
+        {
+            Call(FN_OnUpdatePrefabEntity, prefabEntity);
+        }
+    };
+
+    DefaultProceduralPrefabGroup::DefaultProceduralPrefabGroup()
+    {
+        PrefabGroupEventBus::Handler::BusConnect();
+    }
+
+    DefaultProceduralPrefabGroup::~DefaultProceduralPrefabGroup()
+    {
+        PrefabGroupEventBus::Handler::BusDisconnect();
+    }
+
+    void DefaultProceduralPrefabGroup::Reflect(ReflectContext* context)
+    {
+        if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+        {
+            behaviorContext->EBus<PrefabGroupNotificationBus>("PrefabGroupNotificationBus")
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                ->Attribute(AZ::Script::Attributes::Module, "prefab")
+                ->Handler<PrefabGroupNotificationHandler>()
+                ->Event("OnUpdatePrefabEntity", &PrefabGroupNotificationBus::Events::OnUpdatePrefabEntity);
+
+            behaviorContext->EBus<PrefabGroupEventBus>("PrefabGroupEventBus")
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                ->Attribute(Script::Attributes::Module, "prefab")
+                ->Event("GeneratePrefabGroupManifestUpdates", &PrefabGroupEvents::GeneratePrefabGroupManifestUpdates);
+        }
+    }
+
+    AZStd::optional<PrefabGroupEvents::ManifestUpdates> DefaultProceduralPrefabGroup::GeneratePrefabGroupManifestUpdates(
+        const Scene& scene) const
+    {
+        auto nodeDataMap = CalculateNodeDataMap(scene);
+        if (nodeDataMap.empty())
+        {
+            return {};
+        }
+
+        // compute the filenames of the scene file
+        AZStd::string relativeSourcePath = scene.GetSourceFilename();
+        // the watch folder and forward slash is used in the asset hint path of the file
+        AZStd::string watchFolder = scene.GetWatchFolder() + "/";
```

Comment:
should this use pathutils or the define for the correct path separator, instead of hardcoding to "/"?

Reply:
this is not a path separator, this is an asset path which are always forward slashes. The danger of using a API to "fix it" would be if the API starts to think that back slashes (\) for Windows should be used... they should not be